updatehub/src/settings.rs

```diff
@@ -64,8 +62,8 @@ impl Settings {
     /// Loads the settings from the filesystem. If
     /// `/etc/updatehub.conf` does not exists, it uses the default
     /// settings.
-    pub fn load() -> Result<Self> {
-        let path = std::path::Path::new(SYSTEM_SETTINGS_PATH);
+    pub fn load(path: std::path::PathBuf) -> Result<Self> {
+        let path = std::path::Path::new(&path);
```

Comment:
`path` argument should be `&std::path::Path` and the conversion will be implicit on the deref of wherever this is called.